lib/metasploit/framework/login_scanner/mssql.rb

```diff
@@ -82,6 +82,7 @@ def attempt_login(credential)
               result_options[:status] = Metasploit::Model::Login::Status::SUCCESSFUL
               if use_client_as_proof
                 result_options[:proof] = client
+                result_options[:connection] = client.sock
```

Comment:
Oh that's interesting; you're right that this is the intended API that should've been used 👀 

https://github.com/rapid7/metasploit-framework/blob/c243125612d1c9660aea464dd3d71bf943d3f6d5/lib/metasploit/framework/login_scanner/result.rb#L23-L25

https://github.com/rapid7/metasploit-framework/blob/c243125612d1c9660aea464dd3d71bf943d3f6d5/lib/metasploit/framework/login_scanner/result.rb#L35-L37

Reply:
Ah t'was added by smashery for the WinRM effort - nice :+1:

https://github.com/rapid7/metasploit-framework/commit/39f90d95b1d81b29afab31b653cbd6a231889b6e#diff-48a6460ca79b6024814bd89d6dea6f074376f282d590e3ab7e4b2355922ab27a

Reply:
yup! although I did notice I believe they're using it for the http client rather than just the socket, which I think isn't an issue but I wanted to call it out just in case